### PR TITLE
Code and structure refactoring

### PR DESCRIPTION
**Pull request by Dmytry-Dyachenko**
In this PR:

- Add Javadocs for aggregates and rejections
- Add `checkNotNull` to public static methods
- Increase code coverage to 98%

Previously we used `foreach` with `if` for finding `loan` or `item` positions. In a new functional we use lambdas. As well, we increase the code readability and quality. 

Refactoring:

- Remove long dash after parameters names
- Add a paragraph between the method sence and the description
- Rename methods and variables for more clarity
- Replace `IllegalArgumentException` with `io.spine.util.Exceptions.newIllegalArgumentException`.

Remove:

- `CannotBorrowBook` rejection
- `CannotReserveBook` rejection

Create:

- `NonAvailableBook` rejection
- `BookAlreadyReserved` rejection
- `BookAlreadyBorrowed` rejection

### CODE REVIEW

File: .spine/spine_model.ser [hunk 1]
```diff
@@ -1,3 +1,3 @@
 
-.javaclasses.exlibris.c.aggregate.BookAggregate
-3javaclasses.exlibris.c.aggregate.InventoryAggregate
```
Comment (alexanderkarpetc) [inline .spine/spine_model.ser:3]: Add whitespace

**Review by alexanderkarpetc — CHANGES_REQUESTED** [review]
Add line at the end of the file, change throw according to new rules

**Review by alexanderkarpetc — APPROVED** [review]

**Review by alexanderkarpetc — APPROVED** [review]
LGTM

**Review by PaulAgeyev — APPROVED** [review]
LGTM.

**Pull request merged**